File: CSharp/LeetCode/0400_0499/0459_RepeatedSubstringPattern.cs
namespace LeetCode;

class RepeatedSubstringPattern
{
    public bool RepeatedSubstringPatternCheck(string s)
    {
        for (int len = s.Length / 2; len >= 1; len--)
        {
            if (s.Length % len == 0)
            {
                string pattern = s.Substring(0, len);
                if (string.Concat(Enumerable.Repeat(pattern, s.Length / len)) == s)
                {
                    return true;
                }
            }
        }

        return false;
    }
}

File: JavaScript/0400_0499/0459_RepeatedSubstringPattern.js
function RepeatedSubstringPatternCheck(s) {
	for (let i = Math.floor(s.length / 2); i >= 1; i--) {
		if (s.length % i == 0) {
			let subString = s.substring(0, i);
			let repeatedSubString = subString.repeat(s.length / i);

			if (repeatedSubString === s) {
				return true;
			}
		}
	}

	return false;
}
